ext/openssl/openssl.c

```diff
@@ -3215,18 +3384,26 @@ PHP_FUNCTION(openssl_csr_new)
 						RETVAL_TRUE;
 
 						if (X509_REQ_sign(csr, req.priv_key, req.digest)) {
-							ZVAL_RES(return_value, zend_register_resource(csr, le_csr));
+							object_init_ex(return_value, x509_request_ce);
+							x509_request_obj = Z_X509_REQUEST_P(return_value);
+							x509_request_obj->csr = X509_REQ_dup(csr);
```

Comment:
There are quite a few code branches when `csr` has a non-null value, but is not assigned in the `x509_request_obj`. If we didn't duplicate it here, we should ensure separately that it is not freed when it is part of the return value.

Please correct me if I'm wrong, because there's a big chance I'm missing something. :)

Reply:
The following `csr = NULL` line is what makes sure that it will not get freed. As is, I expect this will cause a memory leak.